netaddr.go

```diff
@@ -720,6 +720,14 @@ func appendHex(b []byte, x uint16) []byte {
 	return append(b, digits[x&0xf])
 }
 
+// appendHexPad appends the fully padded hex string representation of x to b.
+func appendHexPad(b []byte, x uint16) []byte {
+	b = append(b, digits[x>>12])
+	b = append(b, digits[x>>8&0xf])
+	b = append(b, digits[x>>4&0xf])
+	return append(b, digits[x&0xf])
```

Comment:
I'd do this all in a single line. It's almost as readable, and it'll generate much better code.

Reply:
@mdlayher pls